### PR TITLE
[Switch] Fix initial page load animation 

### PR DESCRIPTION
**Pull request by rohanprasadofficial**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [x] android

### Description of changes

This PR targets to fix the animation behavior that was seen on page load itself which was not expected. 
Animation should only occur when the switch changes the state from ON to OFF & vice-versa.  

This also target to clean up some functions and reduce few code. 

Adding the videos below. 

### Verification

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| https://user-images.githubusercontent.com/30728574/213670626-b76b982b-6841-4218-97d0-6cc32aa24f37.mov | https://user-images.githubusercontent.com/30728574/213670678-8380a6c0-7475-49ee-b14e-8600d323b2f8.mov |

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/experimental/Switch/src/__tests__/__snapshots__/Switch.test.tsx.snap [hunk 1]
```diff
@@ -168,7 +168,7 @@ exports[`Switch Disabled 1`] = `
         ],
       },
       "trackBackgroundStyle": Object {
-        "backgroundColor": "rgba(180, 214, 250, 1)",
+        "backgroundColor": "rgba(240, 240, 240, 1)",
```
Comment (rohanprasadofficial) [inline packages/experimental/Switch/src/__tests__/__snapshots__/Switch.test.tsx.snap:171]: This results in the change as before it was taking initial animation value before moving to the correct state , Now it take correct animation value directly.

**Comment by rohanprasadofficial** [comment]
This bug fix targets the completion of all the pending task for switch for Android. 

FYI - @vishnu2410 @ankraj12

File: packages/experimental/Switch/src/useSwitch.ts [hunk 2]
```diff
@@ -37,62 +31,27 @@ export const useSwitch = (props: SwitchProps, animationConfig?: AnimationConfig)
...
+      const toValue = checkedState ? 0 : 1;
+      Animated.timing(trackBackgroundAnimation, {
+        toValue,
+        duration: isInit ? 0 : 500,
+        useNativeDriver: false,
+      }).start();
+
+      setIsInit(false);
+    }
+  }, [checked, checkedState]);
```
Comment (rurikoaraki) [inline packages/experimental/Switch/src/useSwitch.ts:54]: nit: doesn't look like this depends on checked anymore?

**Review by rurikoaraki — APPROVED** [review]

File: packages/experimental/Switch/src/useSwitch.ts (hunk 2 above)
Comment (rohanprasadofficial) [reply packages/experimental/Switch/src/useSwitch.ts:54]: @rurikoaraki , I tried with it but we have removed function with all the animation code in it and will be handling with this function itself.So when also checked state chnages we will need call this.

**Pull request merged**